test/test_suite_main_testcases.hpp

```diff
@@ -792,6 +821,90 @@ PRINTF_TEST_CASE(floating_point_specifiers_precision_and_flags)
+  PRINTING_CHECK("4.0",                     ==, sprintf_, buffer, "%.1lf", 3.999);
+  PRINTING_CHECK("4",                       ==, sprintf_, buffer, "%.0lf", 3.5);
+  PRINTING_CHECK("4",                       ==, sprintf_, buffer, "%.0lf", 4.5);
+  PRINTING_CHECK("3",                       ==, sprintf_, buffer, "%.0lf", 3.49);
+  PRINTING_CHECK("3.5",                     ==, sprintf_, buffer, "%.1lf", 3.49);
+  PRINTING_CHECK("a0.5  ",                  ==, sprintf_, buffer, "a%-5.1lf", 0.5);
+  PRINTING_CHECK("a0.5  end",               ==, sprintf_, buffer, "a%-5.1lfend", 0.5);
+
+  /* %llf for long double */
+  PRINTING_CHECK("3.1415",                  ==, sprintf_, buffer, "%.4llf", 3.1415354l);
+  PRINTING_CHECK("30343.142",               ==, sprintf_, buffer, "%.3llf", 30343.1415354l);
+  PRINTING_CHECK("34",                      ==, sprintf_, buffer, "%.0llf", 34.1415354l);
+  PRINTING_CHECK("1",                       ==, sprintf_, buffer, "%.0llf", 1.3l);
+  PRINTING_CHECK("2",                       ==, sprintf_, buffer, "%.0llf", 1.55l);
+  PRINTING_CHECK("1.6",                     ==, sprintf_, buffer, "%.1llf", 1.64l);
+  PRINTING_CHECK("42.90",                   ==, sprintf_, buffer, "%.2llf", 42.8952l);
+  PRINTING_CHECK("42.895200000",            ==, sprintf_, buffer, "%.9llf", 42.8952l);
+  PRINTING_CHECK("42.8952230000",           ==, sprintf_, buffer, "%.10llf", 42.895223l);
+  PRINTING_CHECK("42.895223123457",         ==, sprintf_, buffer, "%.12llf", 42.89522312345678l);
+  PRINTING_CHECK("42477.371093750000000",   ==, sprintf_, buffer, "%020.15llf", 42477.37109375l);
+  PRINTING_CHECK("42.895223876543",         ==, sprintf_, buffer, "%.12llf", 42.89522387654321l);
+  PRINTING_CHECK(" 42.90",                  ==, sprintf_, buffer, "%6.2llf", 42.8952l);
+  PRINTING_CHECK("+42.90",                  ==, sprintf_, buffer, "%+6.2llf", 42.8952l);
+  PRINTING_CHECK("+42.9",                   ==, sprintf_, buffer, "%+5.1llf", 42.9252l);
+  PRINTING_CHECK("42.500000",               ==, sprintf_, buffer, "%llf", 42.5l);
+  PRINTING_CHECK("42.5",                    ==, sprintf_, buffer, "%.1llf", 42.5l);
+  PRINTING_CHECK("42167.000000",            ==, sprintf_, buffer, "%llf", 42167.0l);
+  PRINTING_CHECK("-12345.987654321",        ==, sprintf_, buffer, "%.9llf", -12345.987654321l);
+  PRINTING_CHECK("4.0",                     ==, sprintf_, buffer, "%.1llf", 3.999l);
+  PRINTING_CHECK("4",                       ==, sprintf_, buffer, "%.0llf", 3.5l);
+  PRINTING_CHECK("4",                       ==, sprintf_, buffer, "%.0llf", 4.5l);
+  PRINTING_CHECK("3",                       ==, sprintf_, buffer, "%.0llf", 3.49l);
+  PRINTING_CHECK("3.5",                     ==, sprintf_, buffer, "%.1llf", 3.49l);
+  PRINTING_CHECK("a0.5  ",                  ==, sprintf_, buffer, "a%-5.1llf", 0.5l);
+  PRINTING_CHECK("a0.5  end",               ==, sprintf_, buffer, "a%-5.1llfend", 0.5l);
```

Comment:
I don't see any checks which try to utilize the greater precision of long doubles - for the case of `long double` being different than double.

Reply:
If we try to utilize the greater precision of long double, we need 
`PRINTF_USE_LONG_DOUBLE_INTERNALLY`
like `PRINTF_USE_DOUBLE_INTERNALLY`
but PRINTF_USE_LONG_DOUBLE_INTERNALLY is not implemented yet,
and implement that will be complicated, so we use print
long double with `double` or `float` currently

only make sure %lf with `long double` won't be messed up

Reply:
So, add a comment to this effect. Also, switch the specified to `%Lf`, not `%llf`.